runtime/shared_common/shrinit.cpp

```diff
@@ -5128,4 +5156,60 @@ j9shr_findGCHints(J9VMThread* currentThread, UDATA *heapSize1, UDATA *heapSize2)
 	return returnVal;
 }
 
+
+/**
+ * Check if the length of the cache file path is greater than the maximum limit.
+ *
+ * @param [in] vm Pointer to the VM structure
+ * @param [in] cacheDirName The cache control directory
+ * @param [in] cacheName The shared cache name
+ * @param [in] cacheType The type of the shared cache
+
+ *
+ * @return 0 if the length of the cache file path is not greater than maximum limit, -1 otherwise.
+ */
+static IDATA
+checkCacheFilePath(J9JavaVM* vm, const char* ctrlDirName, const char* cacheName, U_32 cacheType)
+{
+	char cacheNameWithVGen[J9SH_MAXPATH];
+	char cacheFilePath[J9SH_MAXPATH];
+	char cacheDirName[J9SH_MAXPATH];
+	IDATA rc = 0;
+	PORT_ACCESS_FROM_JAVAVM(vm);	
+	J9PortShcVersion versionData;
+	
+	memset(cacheNameWithVGen, 0, sizeof(cacheNameWithVGen));
```

Comment:
Are the memset's really required?

Reply:
I don't want any garbage value in theses arrays. I prefer keeping the memset's.